crates/autopilot/src/database/auction.rs

```diff
@@ -97,18 +97,30 @@ impl Postgres {
         })
     }
 
+    pub async fn get_next_auction_id(&self) -> Result<dto::AuctionId> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["get_next_auction_id"])
+            .start_timer();
+
+        let mut ex = self.pool.acquire().await?;
+        let id = database::auction::get_next_auction_id(&mut ex).await?;
+        Ok(id)
+    }
+
     pub async fn replace_current_auction(
         &self,
+        id: dto::AuctionId,
         auction: &dto::RawAuctionData,
-    ) -> Result<dto::AuctionId> {
+    ) -> Result<()> {
         let _timer = super::Metrics::get()
             .database_queries
-            .with_label_values(&["replace_current_auction"])
+            .with_label_values(&["insert_auction_with_id"])
             .start_timer();
 
         let data = serde_json::to_value(auction)?;
         let mut ex = self.pool.acquire().await?;
-        let id = database::auction::replace_auction(&mut ex, &data).await?;
-        Ok(id)
+        database::auction::insert_auction_with_id(&mut ex, id, &data).await?;
```

Comment:
To supplement my point, I'd have to dive up to here to understand that `id`is the new auction ID